pkg/policy/snapshots.go

```diff
@@ -0,0 +1,236 @@
+
+	return strings.TrimSpace(query)
+}
+
+func (ce *Executor) ExtractTableNames(ctx context.Context, query string) ([]string, error) {
+	tableNames := make([]string, 0)
+	cleanedQuery := cleanQuery(query)
+
+	explainQuery := fmt.Sprintf("EXPLAIN (FORMAT JSON) %s", cleanedQuery)
+
+	ce.log.Debug("extracting Table names-cleaned", "explainQuery", explainQuery, "raw query", query)
+
+	rows, err := ce.conn.Query(ctx, explainQuery)
+	if err != nil {
+		return tableNames, err
+	}
+	var s string
+	for rows.Next() {
+		if err := rows.Scan(&s); err != nil {
+			ce.log.Error("error scanning into variable", "error", err)
+			return nil, err
+		}
+	}
+
+	var arrayJsonMap []map[string](interface{})
+	err = json.Unmarshal([]byte(s), &arrayJsonMap)
+	if err != nil {
+		ce.log.Error("failed to unmarshal json", "err", err)
+		return tableNames, err
+	}
+
+	flat, err := flatten.Flatten(arrayJsonMap[0], "", flatten.DotStyle)
+	if err != nil {
+		ce.log.Error("failed to flatten json", "err", err)
+		return tableNames, err
+	}
+	for key, val := range flat {
+		if strings.HasSuffix(key, "Relation Name") {
+			ce.log.Debug("Found relation Name", "Table", val)
+			tableNames = append(tableNames, val.(string))
+		}
+		if strings.HasSuffix(key, "Alias") { // Aliases could be query aliases or table aliases (views)
+
+			// Check if query alias, if it is the table will not exist
+			viewQuery, err := ce.checkTableExistence(ctx, val.(string))
+			if err != nil && strings.Contains(err.Error(), "does not exist (SQLSTATE 42P01)") {
+				ce.log.Error("failed to grab query", "err", err, "resp", viewQuery)
+				continue
+			} else if err != nil {
+				return tableNames, err
+			}
+			// If the alias is a view name, then recursively extract the table names from the subquery
+			if viewQuery != "" {
+				viewTables, err := ce.ExtractTableNames(ctx, viewQuery)
+				if err != nil {
+					return []string{}, err
+				}
+				ce.log.Error("Added tables from view", "resp", viewTables)
+				tableNames = append(tableNames, viewTables...)
+			}
+		}
+	}
+
+	if err := rows.Err(); err != nil {
+		ce.log.Error("Error fetching rows", "query", query, "error", err)
+		return tableNames, err
+	}
+	// It is possible that tables are used multiple times so need to dedupe prior to returning
+	return removeDuplicateValues(tableNames), err
+}
+
+func removeDuplicateValues(stringSlice []string) []string {
+	keys := make(map[string]bool)
+	list := []string{}
+	for _, entry := range stringSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
+}
+
+func (ce *Executor) StoreOutput(ctx context.Context, pol *Policy, destination string) (err error) {
+	if !pol.HasChecks() {
+		return errors.New("no checks")
+	}
+	err = ce.createViews(ctx, pol)
+	if err != nil {
+		return fmt.Errorf("failed to create views: %w", err)
+	}
+
+	ef, err := os.OpenFile(fmt.Sprintf("%s/data.csv", destination), os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0777)
```

Comment:
what if file already exists?

Reply:
The likelihood of that happening is near 0 because the `destination` is a path created from a UUID